discord/ui/select.py

```diff
@@ -344,11 +351,19 @@ def is_dispatchable(self) -> bool:
 
     @classmethod
     def from_component(cls, component: SelectMenu) -> Self:
-        return cls(
-            **{k: getattr(component, k) for k in cls.__item_repr_attributes__},
-            custom_id=component.custom_id,
-            row=None,
-        )
+        type_to_cls: Dict[ValidSelectType, Type[BaseSelect[Any]]] = {
+            ComponentType.string_select: Select,
+            ComponentType.user_select: UserSelect,
+            ComponentType.role_select: RoleSelect,
+            ComponentType.channel_select: ChannelSelect,
+            ComponentType.mentionable_select: MentionableSelect,
+        }
+        if component.type not in type_to_cls:
+            raise TypeError(f'Unknown component type {component.type}')
+
+        constructor = type_to_cls[component.type]
```

Comment:
```suggestion
        constructor = type_to_cls.get(component.type, Select)
```

I want this to be mostly crash free.